### ISSUE TITLE
Program crashes when trying to open some types of DDS textures

### ISSUE THREAD

**Issue by Oxicid** [nvpro-samples/nvtt_samples#2]
I wanted to modify the game Witcher 3, and faced with the fact that some textures dds format does not open. Perhaps the implementation of the image itself is proprietary, and to you should not be a claim. But at least the program should report that it does not support this type of images. And not a stupid pop up. (When you try to open goes noise on the speakers).

There are textures in the link:
Texture 'alnus.dds' (Diffuse) opens without problems (NVTT is probably the only software that opens it)

Texture 'alnus_n.dds' (Normal Map) and 'alnus_s.dds' (Specular) do not open, the program crashes when trying to open them. Maybe it has something to do with sRGB modes ? It is possible that another program that pulls textures from the game, breaks the texture.

https://drive.google.com/drive/folders/1Esy2ZBMjccM790nux28qR2l5eomjL9g7?usp=sharing

**Comment by Oxicid** [comment]
Looking inside the file, you can see that it is not saved correctly, there are no problems on the nvtt side ![image](https://user-images.githubusercontent.com/76912541/231087288-2b4aabc6-cd72-4cf3-91e8-fb64f75e124d.png)

**Comment by NBickford-NV** [comment]
Hi @Oxicid, thanks for reporting this issue! I can confirm that nvtt_export crashes on `alnus_n.dds`; I'm reopening the issue as a bug, since one of my goals for nvtt_export is that it should never crash, even if you give it an incorrectly saved file.

What's interesting about `alnus_n.dds` is that it has no specified format: its FourCC code is null, and its bit count is also null. As a result, nvtt_export tries to interpret this as reading from a DX9-style DDS file with bitmasks whose data is 0 bytes long, which breaks.

Looking at the raw data in a hex editor, I think this data was written in BC7 format. So, one way to make it readable in the current version of `nvtt_export` is to set the 8 bytes I've selected starting at +0x50 here to `04 00 00 00 42 43 37 00`:

![image](https://github.com/nvpro-samples/nvtt_samples/assets/57467222/24e3cc1e-471e-4094-9db9-2e1fa7059c35)

Note that this "BC7\0" FourCC code is very nonstandard - I think nvtt_export and possibly DirectXTex support it (and nvtt_export never uses this method to specify BC7 when it writes a file). The standard way is to add a DX10 header that specifies the BC7 DXGI format, which `alnus.dds` does, but I've shown the nonstandard way here because it involves editing fewer bytes.

My plan is to implement a fix for this in the next version of nvtt_export. Additionally, this is a good sign that I should re-run fuzz-testing on nvtt_export's DDS reader.

Thank you for finding and reporting this!

**Comment by NBickford-NV** [comment]
Hi @Oxicid ! I'm happy to say that we've just released Texture Tools Exporter 2023.3.0 with a fix for this crash - you can download it from https://developer.nvidia.com/nvidia-texture-tools-exporter.

Although the DDS reader can't guess the correct format for this file at the moment, it can detect the header isn't valid and suggest what to look at. Here's the error that loading alnus_n.dds now produces:

```
This file is probably not valid: mip 0 declared it contained 34359738368 bytes of data. However, a mip of size 2048 x 4096 x 1 would only contain 134217728 bytes of data using the largest DDS format, RGBA32F (which uses 16 bytes per pixel). Is a DDS format missing from the header of this file?
```

(this is because `dwPitchOrLinearSize` in `alnus_n.dds` was set to `0x00800000`; since its DDS format is missing, the DDS reader interprets `dwPitchOrLinearSize` as a pitch, and detects it is too large.)

We also now have some checks against zero-length mips with a null FourCC code and texture sizes overflowing resource limits, from fuzz-testing.

Thanks again for reporting this!
